lib/puppet/util/windows/taskscheduler2_v1task.rb

```diff
@@ -0,0 +1,559 @@
+
+    full_taskname = Puppet::Util::Windows::TaskScheduler2::ROOT_FOLDER + validate_task_name(task_name)
+
+    result = @task.activate(full_taskname)
+    return nil if result.nil?
+    if @task.compatibility != Puppet::Util::Windows::TaskScheduler2::TASK_COMPATIBILITY_AT && @task.compatibility != Puppet::Util::Windows::TaskScheduler2::TASK_COMPATIBILITY_V1
+      @task.deactivate
+      result = nil
+    end
+
+    result
+  end
+
+  def delete(task_name)
+    
+    @task.delete(validate_task_name(task_name))
+  end
+
+  def run()
+    @task.run(nil)
+  end
+
+  def save(file = nil)
+    raise NotImplementedError unless file.nil?
+
+    @task.save
+  end
+
+  def terminate
+    @task.terminate
+  end
+
+  def machine=(host)
+    # The Puppet scheduledtask provider never calls this method.
+    raise NotImplementedError
+  end
+
+  alias :host= :machine=
+
+  # Sets the +user+ and +password+ for the given task. If the user and
+  # password are set properly then true is returned.
+  #
+  # In some cases the job may be created, but the account information was
+  # bad. In this case the task is created but a warning is generated and
+  # false is returned.
+  #
+  # Note that if intending to use SYSTEM, specify an empty user and nil password
+  #
+  # Calling task.set_account_information('SYSTEM', nil) will generally not
+  # work, except for one special case where flags are also set like:
+  # task.flags = Win32::TaskScheduler::TASK_FLAG_RUN_ONLY_IF_LOGGED_ON
+  #
+  # This must be done prior to the 1st save() call for the task to be
+  # properly registered and visible through the MMC snap-in / schtasks.exe
+  #
+  def set_account_information(user, password)
+    @task.set_principal(user, password)
+  end
+
+  def account_information
+    principal = @task.principal
+
+    principal.nil? ? nil : principal.UserId
+  end
+
+  def application_name
+    action = default_action
+    action.nil? ? nil : action.Path
+  end
+
+  def application_name=(app)
+    action = default_action(true)
+    action.Path = app
+
+    app
+  end
+
+  def parameters
+    action = default_action
+    action.nil? ? nil : action.Arguments
+  end
+
+  def parameters=(param)
+    action = default_action(true)
+    action.Arguments = param
+
+    param
+  end
+
+  def working_directory
+    action = default_action
+    action.nil? ? nil : action.WorkingDirectory
+  end
+
+  def working_directory=(dir)
+    action = default_action
+    action.WorkingDirectory = dir
+
+    dir
+  end
+
+  def priority
+    @task.priority
+  end
+
+  def priority=(value)
+    raise TypeError unless value.is_a?(Numeric)
+
+    @task.priority_value = value
+
+    value
+  end
+
+  # Creates a new work item (scheduled job) with the given +trigger+. The
+  # trigger variable is a hash of options that define when the scheduled
+  # job should run.
+  #
+  def new_work_item(task_name, task_trigger)
+    raise TypeError unless task_trigger.is_a?(Hash)
```

Comment:
Should we create a default exec action at the same we're creating the named task and its trigger? Might be the sensible happy-path use case?